week4/project-final/src/pages/Home.js

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useState } from "react";
+import Search from "../components/Search";
+import Title from "../components/Title";
+import WeatherCards from "../components/WeatherCards";
+import CountryNames from "../components/CountryNames";
+import WeatherDesc from "../components/WeatherDesc";
+import WeatherDetails from "../components/WeatherDetails";
+import UserFeedback from "../components/UserFeedback";
+import kelvinToCelsius from "kelvin-to-celsius";
+import uuid from "uuid/v1";
+import "array.prototype.move";
+import DeleteButton from "../components/DeleteButton";
+
+const HomePage = () => {
+  const [cityList, setCityList] = useState([]);
+  const [status, setStatus] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  async function fetchWeather(city) {
+    try {
+      const res = await fetch(
+        `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}`
+      );
+      setStatus("Loading");
+      if (!res.ok) {
+        throw Error(
+          "Uppss.. An error occurred while fetching the data. Make sure you don't have a typo on your input."
+        );
+      }
+      const weatherData = await res.json();
+
+      const isDuplicatedInput =
+        cityList.filter(c => c.id === weatherData.id).length !== 0;
```

Comment:
Maybe this could win the simplicity reward:
```suggestion
        cityList.some(c => c.id === weatherData.id);
```